src/sys/ioctl/mod.rs

```diff
@@ -145,3 +125,75 @@ extern "C" {
 /// A hack to get the macros to work nicely.
 #[doc(hidden)]
 pub use ::libc as libc;
+
+/// Convert raw ioctl return value to a Nix result
+#[macro_export]
+macro_rules! convert_ioctl_res {
+    ($w:expr) => (
+        {
+            $crate::Errno::result($w)
+        }
+    );
+}
+
+/// Declare a wrapper function around an ioctl.
+#[macro_export]
+macro_rules! ioctl {
+    (bad $name:ident with $nr:expr) => (
+        pub unsafe fn $name(fd: $crate::sys::ioctl::libc::c_int,
```

Comment:
This seems to be overloading the "bad" keyword.  AFAICT, in Linux the difference between _IOR and _IOR_BAD is in the interpretation of the size argument.  However, the bad macro here deals with callers who hard-code the ioctl number.  I know @conradev didn't introduce the bad keyword, but it would be nice to fix